Locked accounts cannot be unlocked

## Expected Behavior

If a user gets locked out then a terraform plan produces this:
```
  ~ okta_user.some_user
      status:           "LOCKED_OUT" => "ACTIVE"
```

Activating the user should be possible.

## Actual Behavior

If a user is locked out but active in the terraform config and then the above plan is applied, the following occurs:

```
Error: Error applying plan:

1 error occurred:
	* okta_user.some_user: 1 error occurred:
	* okta_user.some_user: [ERROR] Error Updating Status for User: The API returned an error: Activation failed because the user is already active
```

## Steps to Reproduce the Problem

1. Have an active user in the terraform code
2. Lock that user out, e.g. too many incorrect password entries
3. Try to plan and apply

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -53,3 +53,4 @@ For Release v3.0.10
 For Release v3.0.11
 
 * Fix ocassional panic when creating a user schema see https://github.com/articulate/terraform-provider-okta/issues/144
+* Users in LOCKED_OUT state are unlocked when config is ACTIVE https://github.com/articulate/terraform-provider-okta/issues/225

--- a/okta/user.go
+++ b/okta/user.go
@@ -498,7 +498,7 @@ func updateGroupsOnUser(u string, g []string, c *okta.Client) error {
 // handle setting of user status based on what the current status is because okta
 // only allows transitions to certain statuses from other statuses - consult okta User API docs for more info
 // https://developer.okta.com/docs/api/resources/users#lifecycle-operations
-func updateUserStatus(u string, d string, c *okta.Client) error {
+func updateUserStatus(uid string, desiredStatus string, c *okta.Client) error {
 	user, _, err := c.User.GetUser(u)
 
 	if err != nil {
@@ -506,19 +506,23 @@ func updateUserStatus(u string, d string, c *okta.Client) error {
 	}
 
 	var statusErr error
-	switch d {
+	switch desiredStatus {
 	case "SUSPENDED":
-		_, statusErr = c.User.SuspendUser(u)
+		_, statusErr = c.User.SuspendUser(uid)
 	case "DEPROVISIONED":
-		_, statusErr = c.User.DeactivateUser(u)
+		_, statusErr = c.User.DeactivateUser(uid)
 	case "ACTIVE":
-		if user.Status == "SUSPENDED" {
-			_, statusErr = c.User.UnsuspendUser(u)
-		} else if user.Status == "PASSWORD_EXPIRED" {
+		switch user.Status {
+		case "SUSPENDED":
+			_, statusErr = c.User.UnsuspendUser(uid)
+			break
+		case "PASSWORD_EXPIRED":
 			// Ignore password expired status. This status is already activated.
 			return nil
-		} else {
-			_, _, statusErr = c.User.ActivateUser(u, nil)
+		case "LOCKED_OUT":
+			_, statusErr = c.User.UnlockUser(uid)
+		default:
+			_, _, statusErr = c.User.ActivateUser(uid, nil)
 		}
 	}
 
@@ -526,7 +530,7 @@ func updateUserStatus(u string, d string, c *okta.Client) error {
 		return statusErr
 	}
 
-	err = waitForStatusTransition(u, c)
+	err = waitForStatusTransition(uid, c)
 
 	if err != nil {
 		return err

--- a/okta/user.go
+++ b/okta/user.go
@@ -499,7 +499,7 @@ func updateGroupsOnUser(u string, g []string, c *okta.Client) error {
 // only allows transitions to certain statuses from other statuses - consult okta User API docs for more info
 // https://developer.okta.com/docs/api/resources/users#lifecycle-operations
 func updateUserStatus(uid string, desiredStatus string, c *okta.Client) error {
-	user, _, err := c.User.GetUser(u)
+	user, _, err := c.User.GetUser(uid)
 
 	if err != nil {
 		return err

